tests/test_crm.py

```diff
@@ -0,0 +1,538 @@
+    fvs = swsscommon.FieldValuePairs([("port","Ethernet0"),("type","dynamic")])
+    tbl.set("Vlan2:52-54-00-25-06-E9", fvs)
+
+    # create vlan
+    tbl = swsscommon.Table(cfg_db, "VLAN", "|")
+    fvs = swsscommon.FieldValuePairs([("vlanid", "2")])
+    tbl.set("Vlan2", fvs)
+
+    # create vlan member
+    tbl = swsscommon.Table(cfg_db, "VLAN_MEMBER", "|")
+    fvs = swsscommon.FieldValuePairs([("tagging_mode", "untagged")])
+    tbl.set("Vlan2|Ethernet0", fvs)
+
+    # update available counter
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY', '999')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_available')
+
+    assert new_used_counter - used_counter == 1
+    assert avail_counter - new_avail_counter == 1
+
+    # update available counter
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY', '1000')
+
+    time.sleep(60)
+
+    # get counters
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_available')
+
+    assert new_avail_counter == avail_counter
+
+
+def test_CrmAcl(dvs):
+
+    db = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+    adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+
+    dvs.runcmd("crm config polling interval 1")
+
+    bind_ports = ["Ethernet0", "Ethernet4"]
+
+    # create ACL table
+    ttbl = swsscommon.Table(db, "ACL_TABLE", '|')
+    fvs = swsscommon.FieldValuePairs([("policy_desc", "test"), ("type", "L3"), ("ports", ",".join(bind_ports))])
+    ttbl.set("test", fvs)
+
+    # create ACL rule
+    rtbl = swsscommon.Table(db, "ACL_RULE", '|')
+    fvs = swsscommon.FieldValuePairs([("priority", "55"), ("PACKET_ACTION", "FORWARD"), ("L4_SRC_PORT", "65000")])
+    rtbl.set("test|acl_test_rule", fvs)
+
+    time.sleep(5*60)
+
+    table_used_counter = getCrmCounterValue(dvs, 'ACL_STATS:INGRESS:PORT', 'crm_stats_acl_table_used')
+    assert table_used_counter == 1
+
+    # get ACL table key
+    atbl = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_ACL_TABLE")
+    acl_tables = [k for k in atbl.getKeys() if k not in dvs.asicdb.default_acl_table]
+    key = "ACL_TABLE_STATS:{0}".format(acl_tables[0].replace('oid:', ''))
+
+    entry_used_counter = getCrmCounterValue(dvs, key, 'crm_stats_acl_entry_used')
+    assert entry_used_counter == 1
+
+    cnt_used_counter = getCrmCounterValue(dvs, key, 'crm_stats_acl_counter_used')
+    assert entry_used_counter == 1
+
+    # remove ACL rule
+    rtbl._del("test|acl_test_rule")
+
+    time.sleep(60)
+
+    entry_used_counter = getCrmCounterValue(dvs, key, 'crm_stats_acl_entry_used')
+    assert entry_used_counter == 0
+
+    cnt_used_counter = getCrmCounterValue(dvs, key, 'crm_stats_acl_counter_used')
+    assert cnt_used_counter == 0
+
+    # remove ACL table
+    ttbl._del("test")
```

Comment:
Did you also try deleting the table without removing the rules and checking for CRM values? Just want to know the result. No need to add the test case here.

Reply:
Yes, I tried it and after deleting only ACL table all related CRM values were correctly updated:
1. Table counter decreased by 1;
2. Associated "per table" counters were decreased to 0 and eventually removed.